flaml/automl.py

```diff
@@ -1108,10 +1108,22 @@ def _validate_data(
             from .data import DataTransformer
 
             self._transformer = DataTransformer()
+
             self._X_train_all, self._y_train_all = self._transformer.fit_transform(
                 X, y, self._state.task
             )
             self._label_transformer = self._transformer.label_transformer
+            if hasattr(self._transformer, "_label_list"):
+                for each_estimator in list(self._state.fit_kwargs_by_estimator.keys()):
+                    self._state.fit_kwargs_by_estimator[each_estimator].update(
+                        {"label_list": self._transformer._label_list}
+                    )
+            if self._state.task == TOKENCLASSIFICATION:
+                for each_estimator in list(self._state.fit_kwargs_by_estimator.keys()):
+                    assert (
+                        "label_list"
+                        in self._state.fit_kwargs_by_estimator[each_estimator]
+                    ), "For the token-classification task, you must either (1) pass token labels; (2) pass id labels and the label list. Please refer to the documentation for more details: [a simple token classification example](https://microsoft.github.io/FLAML/docs/Examples/AutoML-NLP#a-simple-token-classification-example)"
```

Comment:
```suggestion
                    ), "For the token-classification task, you must either (1) pass token labels; or (2) pass id labels and the label list. Please refer to the documentation for more details: https://microsoft.github.io/FLAML/docs/Examples/AutoML-NLP#a-simple-token-classification-example"
```
assert msg is plain text.